### ISSUE TITLE
[$250] Expense, task, room - Unable to create a new line in description field with SHIFT+Enter

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49374]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.37-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Email or phone of affected tester (no customers):** applausetester+kh010901@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to chat.
3. Click + > Submit expense > Manual.
4. Enter amount > Next.
5. Click Description.
6. Press SHIFT+ENTER (enter key for mweb).

## Expected Result:
A new line will be created with SHIFT+ENTER.

## Actual Result:
A new line is not created with SHIFT+ENTER. The description is saved instead.

This issue happens in IOU, task and room description.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/08d53f77-a51b-419f-bc04-f38fd9569f3e

Add any screenshot/video evidence
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021836355229392782754</li>
        <li>Upwork Job ID: 1836355229392782754</li>
        <li>Last Price Increase: 2024-09-18</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 104086887</li>
<li>nkdengineer | Contributor | 104086888</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @DylanDylann</details>

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-control

**Comment by nkdengineer** [comment]
I think the RCA is not the same.

**Comment by cead22** [comment]
Thanks @etCoderDysto. Can you check if you can reproduce this using the branch for this PR https://github.com/Expensify/App/pull/49370?

**Comment by cead22** [comment]
I don't think this needs to block the deploy

**Comment by etCoderDysto** [comment]
> Thanks @etCoderDysto. Can you check if you can reproduce this using the branch for this PR #49370?

Sure! I am checking now...

**Comment by etCoderDysto** [comment]
@cead22 I am able to reproduce the issue on this branch https://github.com/Expensify/App/pull/49370. The issues are not related. 

https://github.com/user-attachments/assets/b4aa827e-f5b6-4096-a688-ff23e889ccdc

**Comment by cead22** [comment]
Thank you! Feel free to submit a proposal

**Comment by Skalakid** [comment]
Hello, after testing this regression comes from `useKeyboardShortcut` hook that was added in `SearchRouter` [here](https://github.com/Expensify/App/pull/48785/commits/27724319043df45394231b96a5d333910b936c14)

**Comment by luacmartins** [comment]
This issue might have the same root cause as https://github.com/Expensify/App/issues/49368

**Comment by luacmartins** [comment]
Ah I see that it was introduced by [this PR](https://github.com/Expensify/App/pull/48785/commits/27724319043df45394231b96a5d333910b936c14). @Kicu we might need to rethink the shortcuts for the SearchRouter

**Comment by Kicu** [comment]
@luacmartins yes I think we do. @Skalakid reached to me internally to discuss this, and I suggested that the fastest solution for now would be to remove these two hooks.
 - SearchRouter is in very early stages and its fine to break keyboard handling for 1-2 days at this point IMO
 - with removal of hooks we unblock main, and then I will clean this up and fix it in a separate PR

**Comment by luacmartins** [comment]
Cool, PR fixing this has been merged https://github.com/Expensify/App/pull/49387

**Comment by DylanDylann** [comment]
@luacmartins Ops, I am on the checklist 😄

**Comment by DylanDylann** [comment]
Anyways, I see the original bug still happen 

https://github.com/user-attachments/assets/cdc0c49c-7805-4024-b8da-18ac9611de94

**Comment by luacmartins** [comment]
Oh weird, I'm seeing the issue again but now it navigates back @Skalakid since you worked on the PR.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
A new line is not created with SHIFT+ENTER. The description is saved instead.

### What is the root cause of that problem?
The markdown input is a `div` component with role `textarea`

<img width="627" alt="Screenshot 2024-09-19 at 12 58 08" src="https://github.com/user-attachments/assets/b81fc1bc-539d-43b9-a5b5-1396abfa7355">

Then `shouldDisableEnterShortcut` is `false`

https://github.com/Expensify/App/blob/f7a4aa13979ae71631df79a6ac181c2e9b08f17f/src/components/Button/index.tsx#L152

and this check is passed because the `nodeName` is `DIV`
https://github.com/Expensify/App/blob/f7a4aa13979ae71631df79a6ac181c2e9b08f17f/src/components/Button/validateSubmitShortcut/index.ts#L13-L17

### What changes do you think we should make in order to solve the problem?

We can add `TEXTAREA` role to `CONST.ROLE`

Or we can return false here if the `eventTarget.role` is `textarea`

```
if (isDisabled || isLoading || eventTarget.nodeName === 'TEXTAREA' || eventTarget.role === 'textarea') {
    return false;
}
```
https://github.com/Expensify/App/blob/f7a4aa13979ae71631df79a6ac181c2e9b08f17f/src/components/Button/validateSubmitShortcut/index.ts#L13-L17

### What alternative solutions did you explore? (Optional)

**Comment by Skalakid** [comment]
Oh, okay, my PR fixed the issue inside the main chat Composer and all other inputs that appear on the main panel. Inputs in task, expense, and room pages are still broken

**Comment by Skalakid** [comment]
@nkdengineer, thanks for the proposal, to make it work, we should add the condition ` eventTarget.role === 'textbox'` not `textarea`. However, I think putting the condition `eventTarget?.contentEditable` would be better because it's more generic and will work with any `contentEditable` element, not just Live Markdown Input. @DylanDylann @luacmartins what do you think about it?

**Comment by nkdengineer** [comment]
> thanks for the proposal, to make it work, we should change the condition to eventTarget.role === 'textbox'

@Skalakid Thanks for your feedback, this is the result when I log the `eventTarget.role` here for markdown input.

https://github.com/Expensify/App/blob/f7a4aa13979ae71631df79a6ac181c2e9b08f17f/src/components/Button/validateSubmitShortcut/index.ts#L13


https://github.com/user-attachments/assets/0f382989-ce9a-4326-9afc-a2fed1941f1c

> However, I think putting the condition eventTarget?.contentEditable would be better because it's more generic and will 
work with any contentEditable element, not just Live Markdown Input

Checking `eventTarget?.contentEditable` may be  incorrect, for example in the image below, I logged the `contentEditable` of a normal `TextInput` and if it still exists then that will prevent the enter to submitting the form if we're focusing on a single line input. Instead, I think we can check `eventTarget.ariaMultiLine` or `(eventTarget?.contentEditable && eventTarget.ariaMultiLine`

**Comment by Skalakid** [comment]
@nkdengineer  Oh okay so we can just add `eventTarget?.contentEditable === 'true'` and it will work even with `TextInput` components as well

**Comment by nkdengineer** [comment]
@Skalakid I think that is also an idea.

**Comment by DylanDylann** [comment]
IMO the both ways looks fine 😄:

1. Using eventTarget.ariaMultiLine
2. Using eventTarget?.contentEditable === 'true'

@luacmartins What is the next action here? I think we can go with @nkdengineer's [proposal](https://github.com/Expensify/App/issues/49374#issuecomment-2360047321)

**Comment by miljakljajic** [comment]
@slafortune thank you for taking this one over as I start my parental leave. 

@luacmartins @cead22 will we be going with @nkdengineer 's proposal? Stevie can get them assigned and take this one forward

**Comment by DylanDylann** [comment]
@luacmartins @cead22 Could you please check [my comment](https://github.com/Expensify/App/issues/49374#issuecomment-2360624466) ?

**Comment by cead22** [comment]
Both work from my perspective, so I'd lean on what you @DylanDylann and @luacmartins think is best

**Comment by DylanDylann** [comment]
@slafortune This is ready for payment. The PR is deployed to production on 1/10

**Comment by slafortune** [comment]
@DylanDylann role - contributor - Paid $250 via upworks
@nkdengineer Role - C+ - Paid $250 via upworks
